Sources/MapboxMaps/Viewport/Transitions/Default/CameraOptionsComponent.swift

```diff
@@ -0,0 +1,21 @@
+internal protocol CameraOptionsComponentProtocol {
+    var cameraOptions: CameraOptions { get }
+    func updated(with cameraOptions: CameraOptions) -> CameraOptionsComponentProtocol?
+}
+
+internal struct CameraOptionsComponent<T>: CameraOptionsComponentProtocol {
+    internal let keyPath: WritableKeyPath<CameraOptions, T?>
```

Comment:
does it have to be a key path to an optional property of `CameraOptions`?  I see that CameraOptions has everything as Optional now, but if in the future we decide to add new property that is non-optional then this will not work for those properties.

Reply:
It wouldn't have to be. Doing it this way allows `value` (on the next line) to be non-optional.

The `*Options` naming pattern in GL Native typically indicates a type where all of its fields are optional. You pass instances to a setter and non-nil fields are updated and any nil fields are not modified. Given that, it's probably unlikely that we'll add non-optional fields to `CameraOptions` in the future.